content/articles/spring-boot-caching/index.md

```diff
@@ -0,0 +1,332 @@
+Cache reduces the number of database reads resulting in reduced cost of operation and improved application speed. Spring Boot provides an abstraction layer for cache implementation. 
+		
+
+In this article, we will learn how to implement a cache in a Spring Boot REST application using Ehcache as the cache provider.
+		
+
+### Table of contents
+- [Table of contents](#table-of-contents)
+- [Prerequisites](#prerequisites)
+- [Project setup](#project-setup)
+- [Domain layer](#domain-layer)
+- [Repository layer](#repository-layer)
+- [Service layer](#service-layer)
+- [Controller layer](#controller-layer)
+- [Configuaration layer](#configuaration-layer)
+- [Adding data into the Cache](#adding-data-into-the-cache)
+- [Updating data in the cache](#updating-data-in-the-cache)
+- [Deleting data from the cache](#deleting-data-from-the-cache)
+- [Conclusion](#conclusion)
+		
+### Prerequisites
+1. [JDK](https://www.oracle.com/java/technologies/javase/javase-jdk8-downloads.html) installed on your computer.
+2. Knowledge of Spring Boot.
+		
+
+### Project setup
+We are going to use [spring initializr](https://start.spring.io/) to bootstrap our application.
+		
+
+1. Visit [spring initializr](https://start.spring.io/), input the project name as `SpringBootCaching`.
+2. Add `Spring Web`, `H2 database`, `Spring data JPA`, and `Lombok` as project dependencies.
+3. Click on generate project button to download the project boilerplate code as a zip file.
+4. Extract the zip file and open the uncompressed file in your favorite IDE.
+5. In the `pom.xml` file in the dependencies section, add the dependencies below.
+ ```xml
+<dependencies>
+  <dependency
+      <groupId>org.springframework.boot</groupId>
+      <artifactId>spring-boot-starter-cache</artifactId>
+   </dependency>
+
+   <dependency>
+      <groupId>javax.cache</groupId>
+      <artifactId>cache-api</artifactId>
+   </dependency>
+
+   <dependency>
+      <groupId>org.ehcache</groupId>
+      <artifactId>ehcache</artifactId>
+    </dependency>
+</dependencies>
+``` 
+The above dependencies add `spring-boot-starter-cache`, an abstraction layer implemented by Spring for cache manipulation and `Ehcache` is the cache provider we will be using in our application.
+		
+
+#### Domain layer
```

Comment:
Before you dive into building the project. State the objective - what are you building? what are going to learn by building this?